sql/src/main/java/io/crate/operation/tablefunctions/GenerateSeriesFunction.java

```diff
@@ -0,0 +1,169 @@
+
+            assert numRows>=0 : "invalid combination: cannot reach "+stop+" from "+start+" by stepping "+step;
+
+            return new Bucket() {
+                final Object[] cells = new Object[numCols];
+                final RowN row = new RowN(cells);
+
+                @Override
+                public int size() {
+                    return numRows+1;
+                }
+
+                @Override
+                public Iterator<Row> iterator() {
+                    return new Iterator<Row>() {
+
+                        int currentRow = 0;
+
+                        @Override
+                        public boolean hasNext() {
+                            return currentRow < size();
+                        }
+
+                        @Override
+                        public Row next() {
+                            if (!hasNext()) {
+                                throw new NoSuchElementException("No more rows");
+                            }
+                            cells[0] = start + currentRow*step;
+                            currentRow++;
+                            return row;
+                        }
+
+                        @Override
+                        public void remove() {
+                            throw new UnsupportedOperationException("remove is not supported for " +
+                                GenerateSeriesFunction.class.getSimpleName() + "$iterator");
+                        }
+                    };
+                }
+            };
+        }
+
+        @Override
+        public TableInfo createTableInfo(ClusterService clusterService) {
+            int noElements = info.ident().argumentTypes().size();
+            Map<ColumnIdent, Reference> columnMap = new LinkedHashMap<>(noElements);
+            Collection<Reference> columns = new ArrayList<>(noElements);
+
+            ColumnIdent columnIdent = new ColumnIdent("generate_series");
+            DataType dataType = DataTypes.LONG;
+            Reference reference = new Reference(
+                new ReferenceIdent(TABLE_IDENT, columnIdent),
+                RowGranularity.DOC, dataType);
+
+            columns.add(reference);
+            columnMap.put(columnIdent, reference);
+
+            final String localNodeId = clusterService.localNode().getId();
+            return new StaticTableInfo(TABLE_IDENT, columnMap, columns, Collections.emptyList()) {
+                @Override
+                public RowGranularity rowGranularity() {
+                    return RowGranularity.DOC;
+                }
+
+                @Override
+                public Routing getRouting(WhereClause whereClause, @Nullable String preference) {
+                    return Routing.forTableOnSingleNode(TABLE_IDENT, localNodeId);
+                }
+            };
+        }
+    }
+
+    public static void register(TableFunctionModule module){
+        module.register(NAME, new BaseFunctionResolver(
```

Comment:
Would it be possible to register the FunctionImplementation instance directly? There are only 2 signatures which make sense: `Long -> Long -> Long` and `Long -> Long`.